src/transport/CryptoContext.h

```diff
@@ -165,7 +163,9 @@ class DLL_EXPORT CryptoContext
     SessionRole mSessionRole;
 
     bool mKeyAvailable;
-    CryptoKey mKeys[KeyUsage::kNumCryptoKeys];
+    Crypto::AesKey mEncryptionKey;
+    Crypto::AesKey mDecryptionKey;
+    Crypto::AesKeyByteArray mAttestationChallenge;
```

Comment:
Why not just use SymmetricKeyContext everywhere, rather than mix new AesKey and SymmetricKeyContext?

Reply:
Abstracting at the key level, but without operations using the keys being present, just means that you have "opaque keys". Many different crypto HAL implementations consider the underlying key type to be opaque, but not interchangeable. The AesKey here is used in non-Aes contexts, such as AesKeyByteArray, which is actually just a secret, not even an AES key.

If we used AesKey as written in this PR for the group keys, then the Crypto Backends would need to know where to go get the keys. The would need to be "initialized" at some point and not ephemeral. The GroupDataProvider maintains the keys in storage, and allows getting a SymmetricKeyContext. However, the SymmetricKeyContext allows the keys to be used "at time of use", by reaching in GgroupDataProvider just in time, rather than needing to carry the key outright the whole time, or otherwise have an always available key handle. In that case, it actually limits the contact surface of the key compared to moving to Crypto:AesKey.

Reply:
`AesKey` is used as `AesKeyByteArray` only for platforms that choose that representation. I admit it is currently often initialized from `AesKeyByteArray` and I planned to change it, but didn't want to do too much work in a single PR.

It's a good point that it would be good not to assume that `CryptoContext` lives forever, but `SymmetricKeyContext` doesn't change it by itself, too. It requires `GroupDataProvider` to create the context. So either we use `CryptoContext + AesKeys` or `SymmetricKeyContext` we need some sort of new `SymmetricKeystore` interface that would work for both session and group keys (because the current design that forces me to reimplement `GroupDataProvider` just to store the keys securely seems like too tight coupling to me). Would you agree?

Then the question whether we want to go with AesKeys or SymmetricKeyContext is the question whether we want to keep the current crypto primitives like AES_CCM_encrypt in the common interface and write as much common code as possible out of this building blocks, or split at the SymmetricKeyContext level and probably drop the low-level crypto primitives.